### ISSUE TITLE
The duplicate facet does not take into account other facets

### ISSUE THREAD

**Issue by ettorerizza** [OpenRefine/OpenRefine#1302]
Original user question : https://groups.google.com/forum/#!topic/openrefine/JDpix4g2Jc4

Here is an example dataset : 

```
A,B
1,2
3,2
3,2
1,6
3,7
1,7
```

If I make a "text facet" on column A and I select the value 3, a "duplicate facet" on column B should send me only 2 duplicates, not 3. Bug or feature?

![screenshot-127 0 0 1-3333-2017-11-04-12-54-23-292](https://user-images.githubusercontent.com/6286410/32405153-bd1e863e-c15f-11e7-897f-aa328037eea5.jpeg)

![screenshot-127 0 0 1-3333-2017-11-04-12-55-15-295](https://user-images.githubusercontent.com/6286410/32405154-c240c2da-c15f-11e7-8fc6-c0410ff3ef9c.jpeg)

**Comment by thadguidry** [comment]
It's a feature and works on single column. I think we already have an issue somewhere to provide a unique rows facet that would do what your asking.  If not, please create one.

**Comment by thadguidry** [comment]
@eswright
Let's look at the GREL expression in the B facet picture above.  It is asking for this in layman's terms:

(Show me a count of rows in Column B where the value ) is greater than 1

`facetCount(value, 'value', 'B') > 1`

Now let's look at the data rows that are shown with the A facet selected on 3 (clicked and shows orange color)

OpenRefine's data structure grid looks like this in that picture:
column B
row value 2
row value 2
row value 7

Now run the layman terms sentence in your head... what is the count or number of rows in Column B that has a value greater than 1 ?
If you arrived at a result of 3, then the Facet on B column showing a `true 3` seems to be correct and accurate.  3 rows are matching that expression.

But I love pictures to explain better of what facetCount() really does and how it works.  facetCount() by the way is what we use for the Duplicates Facet counting and it simply takes `'value'` in the normal case as shown in the above picture in this issue, but you could override this to say "Only show me duplicate rows that have a value of 7" like I have done so in this picture (notice how many rows are true that have a value of 7?  then what do you think the facetCount() result would be for "true" ?  If you guessed 2, you are correct !!! , there are 2 rows that equally have a value of 7 :

![duplicatesfacetlogic](https://user-images.githubusercontent.com/986438/45855692-e09b3b80-bd16-11e8-87d9-1e7bf9665b51.PNG)

**Comment by wetneb** [comment]
Adding a "filtered duplicates facet" also brings in more issues - it would mean that the order of facets would need to matter (otherwise, with two filtered duplicates facets, you get a circular definition). So this is prone to introduce bugs and unexpected behaviours.

**Comment by thadguidry** [comment]
@wetneb yes, but we could probably put some controls around that... for instance, always making filtered duplicates facet a LAST OUT facet.  In a hypothetical new UI, the "filtered duplicates facet" may not even be part of the regular facets views, where we might want to highlight to users that this is always a FINAL LAYER that is applied (David H. and I talked about final layers in the past, just like Photoshop does layer ordering with a last layer).  How we present these always final layers...could be a UI experiment or Proof of Concept for a budding UI intern or Google Summer Project.

**Comment by msaby** [comment]
Wow... I spend hours working around this, and just discover it is a feature ;-)
I really need use a filtered duplicate facet for a project, and don't know how to do that :/

**Comment by tfmorris** [comment]
This was closed as "Won't Fix" but is also tagged with the 4.0 milestone, which I don't understand.

**Comment by thadguidry** [comment]
Thanks @tfmorris for spotting the mislabeling.  Removed 4.0 label and added facet
